spring-boot-project/spring-boot-docs/src/main/asciidoc/spring-boot-features.adoc

```diff
@@ -5587,6 +5587,39 @@ The following component creates a listener endpoint on the `someTopic` topic:
 	}
 ----
 
+[[boot-deatures-kafka-streams]]
+==== Kafka Streams
+
+Spring for Apache Kafka provides a factory bean to create a `StreamsBuilder` object and
+manage the lifecycle of its streams; the factory bean is created when
+`@EnableKafkaStreams` is present on a `@Configuration` class.
```

Comment:
I can see the auto-configuration enables that so something is not consistent. Perhaps the auto-configuration should check for the presence of `@EnableKafkaStreams` if that makes sense? (We don't do that for Kafka so perhaps that's the wrong idea).

Reply:
My mistake; I originally did not add `@EnableKafkaStreams` in the auto config because of the problem with auto-starting the factory bean.

I later added the autoStartup auto-config to prevent the FB from starting unless the user explicitly sets it, but failed to revisit the docs.

Which do you prefer - requiring a user-specified `@EnableKafkaStreams` or this autoStartup dance? I think I would prefer that the user adds the annotation, but that seems inconsistent with other practices.

Reply:
I don't know really. I'd like this to be consistent. On the other hand, I don't know if adding `kafka-streams` is enough to trigger that behaviour. Since adding `spring-kafka` is enough for the kafka use case, I'd be tempted to say that's the same here (you need `spring-kafka` and `kafka-streams` anyway).

Any potential clash with Spring Cloud Stream and the kafka streams story there?

Reply:
I'll stick with the presence of the jar for consistency (and fix the configurer).

I will reach out to the SCSt team to make them aware of this.